server/src/main/java/org/elasticsearch/transport/RemoteClusterConnection.java

```diff
@@ -582,43 +540,50 @@ public ClusterStateResponse read(StreamInput in) throws IOException {
 
             @Override
             public void handleResponse(ClusterStateResponse response) {
-                try {
-                    if (remoteClusterName.get() == null) {
-                        assert response.getClusterName().value() != null;
-                        remoteClusterName.set(response.getClusterName());
-                    }
-                    try (Closeable theConnection = connection) { // the connection is unused - see comment in #collectRemoteNodes
-                        // we have to close this connection before we notify listeners - this is mainly needed for test correctness
-                        // since if we do it afterwards we might fail assertions that check if all high level connections are closed.
-                        // from a code correctness perspective we could also close it afterwards. This try/with block will
-                        // maintain the possibly exceptions thrown from within the try block and suppress the ones that are possible thrown
-                        // by closing the connection
-                        cancellableThreads.executeIO(() -> {
-                            DiscoveryNodes nodes = response.getState().nodes();
-                            Iterable<DiscoveryNode> nodesIter = nodes.getNodes()::valuesIt;
-                            for (DiscoveryNode n : nodesIter) {
-                                DiscoveryNode node = maybeAddProxyAddress(proxyAddress, n);
-                                if (nodePredicate.test(node) && connectionManager.size() < maxNumRemoteConnections) {
-                                    try {
-                                        // noop if node is connected
-                                        PlainActionFuture.get(fut -> connectionManager.connectToNode(node, null,
-                                            transportService.connectionValidator(node), ActionListener.map(fut, x -> null)));
-                                    } catch (ConnectTransportException | IllegalStateException ex) {
+                if (remoteClusterName.get() == null) {
```

Comment:
This block is dead code since we're always creating the `SniffClusterStateResponseHandler` in this callback https://github.com/elastic/elasticsearch/pull/44825/files#diff-d390b91a2cae82a4f90f1386f8428695R473 that already does the same thing?

Reply:
good point. The only difference I see is that this does it a bit earlier. I have therefore opted to remove the one in SniffClusterStateResponseHandler.